Updating Salesforce Readme to add clarifying language on API Access. (#6715)
PipedreamHQ/pipedream: components/salesforce_rest_api/README.md

> You can install the Pipedream Salesforce app in the [Accounts](https://pipedream.com/accounts) section of your account, or directly in a workflow.

Reviewer: "you can connect your Salesforce account" is more accurate than "install the Pipedream Salesforce app", right? Or is that maybe that is more accurate in SF lingo?

Author: I think "you can connect your Salesforce account" sounds better since we're still on Pipedream vs. if we were to get Pipedream app listed within Salesforce's App Exchange Marketplace. Will fix this - thanks for the callout.

## API Access

> Please note that only certain [editions](https://help.salesforce.com/s/articleView?id=000385436&type=1) of Salesforce are granted API Access: 

Reviewer 2: ```suggestion
Please note that only certain [editions](https://help.salesforce.com/s/articleView?id=000385436&type=1) of Salesforce have API access: 
```

Reviewer 2: Changed "are granted" to "have" for active voice

**Editions with API Access**
- Enterprise Edition
- Unlimited Edition
- Developer Edition
- Performance Edition
 
**Editions without API Access**
- Group Edition
- Essentials Edition

> - Professional Edition (can be purchased as an add-on) 

Reviewer 2: ```suggestion
- Professional Edition (can be purchased as an add-on)
```

Reviewer 2: Removed extra space at the end

> If you are on a Salesforce edition with API Access, you will need to enable it at the user-profile level under **Administrative Permissions** -> **"API-Enabled"**

Reviewer 2: ```suggestion
If you are using a Salesforce edition with API access, you need to enable it at the user-profile level under **Administrative Permissions** -> **"API Enabled"**
```

Reviewer 2: Changed "will need" to "need" for active voice; changed "on" to "using" for clarity
